server/src/test/java/org/elasticsearch/action/bulk/TransportBulkActionIngestTests.java

```diff
@@ -225,7 +232,21 @@ public void setupAction() {
             return null;
         }).when(clusterService).addStateApplier(any(ClusterStateApplier.class));
         // setup the mocked ingest service for capturing calls
-        ingestService = mock(IngestService.class);
+        ingestService = spy(
```

Comment:
`hasIngestPipelines()` now depends on the `IngestService` state. We use the real implementation but spy on `executeBulkRequest()` method for tests assertions.